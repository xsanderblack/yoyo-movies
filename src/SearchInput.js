import React, { Component } from 'react';

export default class SearchInput extends Component {
  onChange = (e) => {
    this.props.onChange(e.target.value);
  };

  onSubmit = (e) => {
    e.preventDefault();
    this.props.onSubmit();
  };

  componentDidMount() {
    if (this.input) {
      this.input.focus();
    }
  };

  render() {
    const { searchTerm } = this.props;
    return (
      <form onSubmit={this.onSubmit}>
        <input
          type='text'
          value={searchTerm}
          onChange={this.onChange}
          ref={node => this.input = node}
        />
        <button
          type='submit'
          disabled={searchTerm.length < 2}
        >
          Search
        </button>
      </form>
    )
  }
};
