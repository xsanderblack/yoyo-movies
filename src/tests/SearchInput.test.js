import React from 'react';
import ReactDOM from 'react-dom';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import SearchInput from '../SearchInput';

it('SearchInput renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SearchInput searchTerm='test' />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('should render SearchInput correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<SearchInput searchTerm='test' />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
