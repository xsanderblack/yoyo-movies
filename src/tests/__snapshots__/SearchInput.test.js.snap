// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchInput correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="test"
  />
  <button
    disabled={false}
    type="submit"
  >
    Search
  </button>
</form>
`;
